.github/workflows/centraldb_intergration_test.yaml

```diff
@@ -51,8 +56,13 @@ jobs:
         cd components/centraldashboard/manifests
         kubectl create ns kubeflow
 
-        export CURRENT_CENTRALDB_IMG=docker.io/kubeflownotebookswg/centraldashboard:latest
-        export PR_CENTRALDB_IMG=${{env.IMG}}:${{env.TAG}}
+        kustomize build overlays/kserve | kubectl apply -f -
+
+        DEPLOYMENT=centraldashboard
+        CONTAINER=centraldashboard
+
+        IMG=${{env.IMG}}:${{env.TAG}}
+        kubectl patch deployment $DEPLOYMENT -n kubeflow --patch \
+          '{"spec": {"template": {"spec": {"containers": [{"name": "'"$CONTAINER"'","image": "'"$IMG"'"}]}}}}'
```

Comment:
@kimwnasptd I don't like that this applies the manifests and then patches them after, this will result in a generation of the pod that fails to run, quickly followed by one with the correct image.

I don't understand what was not working with the old `sed` approach, which modified the manifests before applying them?

Reply:
> @kimwnasptd I don't like that this applies the manifests and then patches them after, this will result in a generation of the pod that fails to run, quickly followed by one with the correct image.

Why is this a hard problem? The deployment will be patched immediately, and we wait for the Deployment object either way

> I don't understand what was not working with the old sed approach, which modified the manifests before applying them?

Previously the flow was:
1. build the manifests
    1. These manifests in `master` would have `latest` tag in images
    2. In a release branch we update the tag, so it would be something like `v1.8.0-rc.0`
3. `sed` was used change the image with tag `latest`, to `${{env.TAG}}`
4. The above worked for `master`, which has tag `latest` but doesn't for release branches

So the script works for `master`, because `latest` image tag is used in manifests, but fails in release branches which use a different tag in manifests.

Reply:
See https://github.com/kubeflow/kubeflow/pull/7252#issuecomment-1707359973